### PR TITLE
OJ-1995: Added boilerplate for integration-tests

### PR DESCRIPTION
**Pull request by rahulGDS**
### What changed

* Code refactoring
* Boilerplate added for integration tests
* Added Github Actions

### Why did it change
To support testing and deployment of OTG

### Issue tracking

- [OJ-1995](https://govukverify.atlassian.net/browse/OJ-1995)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[OJ-1995]: https://govukverify.atlassian.net/browse/OJ-1995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: integration-tests/setEnvVars.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
-process.env.AWS_SECRET_ACCESS_KEY = "local"; //pragma: allowlist secret
+process.env.AWS_ACCOUNT_ID = "123456789012"; //pragma: allowlist secret
 process.env.SFN_MOCK_CONFIG = "/home/stepfunctionslocal/MockConfigFile.json";
 process.env.AWS_DEFAULT_REGION = "local";
 process.env.AWS_ACCESS_KEY_ID = "local";
+process.env.AWS_SECRET_ACCESS_KEY = "local"; //pragma: allowlist secret
+process.env.AWS_SESSION_TOKEN = "local";
+
+process.env.STATE_MACHINE_ARN =
+  "arn:aws:states:eu-west-2:404250751813:stateMachine:pdv-matching-NinoCheck";
```
Comment (surajkumar) [inline integration-tests/setEnvVars.js:9]: Change this to reflect OTG

File: integration-tests/setEnvVars.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+process.env.STATE_MACHINE_ROLE =
+  "arn:aws:iam::404250751813:role/pdv-matching-NinoCheckStateMachineRole-F6BSBWOOSN15";
+
+process.env.NINO_ATTEMPTS_TABLE = "pdv-matching-nino-attempts";
```
Comment (surajkumar) [inline integration-tests/setEnvVars.js:13]: Remove these nino specific env vars

File: integration-tests/tests/.DS_Store
Comment (surajkumar) [inline integration-tests/tests/.DS_Store:1]: Delete this file

File: integration-tests/tests/aws/.DS_Store
Comment (surajkumar) [inline integration-tests/tests/aws/.DS_Store:1]: Delete this file

File: integration-tests/tests/aws/generate-totp/.DS_Store
Comment (surajkumar) [inline integration-tests/tests/aws/generate-totp/.DS_Store:1]: Delete this file

File: integration-tests/tests/mocked/.DS_Store
Comment (surajkumar) [inline integration-tests/tests/mocked/.DS_Store:1]: Delete this file

File: integration-tests/tests/mocked/generate-totp/.DS_Store
Comment (surajkumar) [inline integration-tests/tests/mocked/generate-totp/.DS_Store:1]: Delete this file

File: integration-tests/tests/mocked/generate-totp/docker-compose.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ version: "3.8"
 
 services:
   step_function_local:
-    container_name: "stepfunction_local"
+    container_name: "otg-generate"
```
Comment (surajkumar) [inline integration-tests/tests/mocked/generate-totp/docker-compose.yaml:5]: can you make this name consistent with the others

**Review by surajkumar — CHANGES_REQUESTED** [review]

**Review by surajkumar — APPROVED** [review]

**Review by jkunle — APPROVED** [review]

**Comment by jkunle** [comment]
Please try and re-organise the commits, so it is a lot easier to review. I think this could be 3 different PR's or 3 commits:
• Code refactoring
• Boilerplate added for integration tests
• Added Github Actions

**Pull request merged**